### ISSUE TITLE
Warning when using command dev:module:rewrite:conflicts

### ISSUE THREAD

**Issue by jhoelzl** [netz98/n98-magerun#523]
I have installed the latest n98-magerun version.
When i enter the command
```
php n98-magerun.phar dev:module:rewrite:conflicts 
```

i get all rewrites correctly as output. However, in var/log/system.log i find this error:

 > ERR (3): Warning: is_a() expects exactly 2 parameters, 3 given  in phar:///magento/n98-magerun.phar/src/N98/Magento/Command/Developer/Module/Rewrite/ConflictsCommand.php on line 152

Any suggentions or could this be caused by a third-party module? Thanks!

**Comment by pocallaghan** [comment]
This is a compatibility issue, the third argument of `is_a` is only available on PHP `5.3.9` or greater. I guess the solution will depend on what versions of PHP n98-magerun wishes to support.

**Comment by jhoelzl** [comment]
OK, thanks!

**Comment by tkn98** [comment]
No decision made yet, that was just a first comment. Thanks for bringing the issue up.

Line in question is https://github.com/netz98/n98-magerun/blob/master/src/N98/Magento/Command/Developer/Module/Rewrite/ConflictsCommand.php#L152

`is_subclass_of` might be viable here.

Also when touching that code the `count($classes)` in the `for` expression middle condition is fishy.

`array_reverse` doesn't preserve keys so a simple `forach` can do it. Also the Exception catching is unexplained. So well, there is some stuff to do there :)

**Comment by tkn98** [comment]
@steverobbins: I reviewed your changes and I'm unsure about the fact that with this use the first operand could be string and therefore the first operand of the instanceof operation would cause a `FALSE` while it previously evaluated to `TRUE`.

https://eval.in/private/a751d74c753cfb

**Comment by steverobbins** [comment]
@tkn98 You are correct.  See updated PR.

**Comment by tkn98** [comment]
@steverobbins Thanks for that.

But sorry I haven't commented yesterday as the moment I reviewed it I wasn't sure on my own and needed a talk with a co-worker first.

Can you use reflection for that? And can you extract the check in a (private) method of it's own?

**Comment by steverobbins** [comment]
@tkn98 Let me know if this is what you're looking for.

**Comment by tkn98** [comment]
@steverobbins: I've left some comments from code-review. Also Could you imagine writing a regression test for this issue? E.g. checking for PHP versions below 5.3.9?

**Comment by steverobbins** [comment]
@tkn98 I'll take a crack at it.  I think this involves creating/faking a rewrite conflict.  I'm still a TDD novice so any tips at how to go about this are appreciated.

**Comment by tkn98** [comment]
I see a stalling PR #528, reviewed it and think that the overall code needs to be refactored and the suggested PR is too complicate. Using new is a no-go.

**Comment by cmuench** [comment]
@steverobbins @tkn98  We cannot instanciate any model. This could produce undefined behaviors and support tickets.

**Comment by tkn98** [comment]
@jhoelzl: Sorry for late bump, do you still have this problem (we have not fixed it, so technically it still should exist) and if so can you share the PHP version? Was this related perhaps to Redhat or some other distro that is backporting?

**Comment by jhoelzl** [comment]
Hi @tkn98, yes i still have the issue. My system details are:

* Linux version 2.6.32-504.12.2.el6.x86_64 (mockbuild@c6b9.bsys.dev.centos.org) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-11) (GCC) ) #1 SMP Wed Mar 11 22:03:14 UTC 2015
* PHP 5.3.3

**Comment by steverobbins** [comment]
Can we revisit why instantiating a model isn't acceptable?  The tests pass ¯\_(ツ)_/¯

Or is it an issue with untested 3rd party classes?  Maybe add a class exists check and warn if something wasn't able to be checked because a class was missing.

**Comment by pocallaghan** [comment]
@steverobbins presumably because instantiation can/will cause code to run (triggered by the constructor), which as @cmuench mentioned will cause undefined behaviour.

**Comment by AydinHassan** [comment]
Maybe we could use `ReflectionClass::newInstanceWithoutConstructor` ?

**Comment by cmuench** [comment]
@AydinHassan ```ReflectionClass::newInstanceWithoutConstructor``` is available since PHP 5.4.
We currently support 5.3.

**Comment by AydinHassan** [comment]
For another day, but we maybe we should consider dropping that, EOL and all that :smile:

**Comment by cmuench** [comment]
Yes, but Magento should drop the support first.  https://github.com/OpenMage/magento-mirror/blob/magento-1.9/index.php#L27

**Comment by AydinHassan** [comment]
@cmuench We could consider [doctrine/instantiator](https://github.com/doctrine/instantiator) that is basically a polyfill for  `ReflectionClass::newInstanceWithoutConstructor` on `5.3`. It is a tiny library.

**Comment by ktomk** [comment]
**ReflectionClass** should allow this natively in 5.3. Looking for ctor should not be option #1.
